sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1299,6 +1299,70 @@ object functions {
    */
   def toRadians(columnName: String): Column = toRadians(Column(columnName))
 
+  /**
+   * Returns the first argument-base logarithm of the second argument.
+   *
+   * @group math_funcs
+   * @since 1.4.0
+   */
+  def logarithm(l: Column, r: Column): Column = Logarithm(l.expr, r.expr)
```

Comment:
let's name the 1st argument as "base", and the second as "a".

and also only keep the following 2 versions:

log(base: Double, a: Column)

log(base: Double, a: String)